infer/src/bufferoverrun/bufferOverrunModels.ml

```diff
@@ -127,13 +127,33 @@ let malloc ~can_be_zero size_exp =
     let offset, size = (Itv.zero, Dom.Val.get_itv length) in
     let represents_multiple_values = not (Itv.is_one size) in
     let allocsite =
-      Allocsite.make pname ~node_hash ~inst_num:0 ~dimension:1 ~path ~represents_multiple_values
+      Allocsite.make pname ~caller_pname:(Dom.Mem.get_proc_name mem) ~node_hash ~inst_num:0
```

Comment:
Oops. Really it is. The semantics of `pname` in `ModelEnv.t` is ambiguous depending on where it is constructed. 😮  Good catch!

Reply:
(Unresolved the conversation, just not to miss comments. Feel free to resolve the conversation.)

Reply:
> it could make more sense that adding caller's proc name to ModelEnv, rather than OndemandEnv (which is in Dom.Mem), since the ondemand env is the data for evaluating parameter values on-demand.

I agree. The problem is that we are still using proc name in our fork also when evaluating parameter values and when importing locations and values from callees to callers. It is used to handle Ada's "uplevel global variables" (Ada makes it possible to define nested subprograms and an uplevel global variable is a variable defined in some enclosing subprogram).

Indeed, we switched to using closures (captured variables) to represent these so this code is no longer needed. That being said, I agree, but I would prefer to first remove this code, move callers's proc name to ModeEnv in our fork and then create a PR with a backport in your repo.

Reply:
> I would prefer to first remove this code, move callers's proc name to ModeEnv in our fork and then create a PR with a backport in your repo.

Sounds great. Thank you.